addons/es-modules.js

```diff
@@ -5,20 +5,28 @@ module.exports = {
         'sourceType': 'module',
     },
     'plugins': [
-        'prefer-import',
+        'import',
     ],
     'rules': {
         // Rule to control usage of strict mode directives
         // comment: code inside es6 modules is under strict mode by default, so it's not necessary
         'strict': [2, 'never'],
         // Rule to warn the use of import when the es6-modules addon is activated.
         // comment: This should be a warning instead of an error because there are still cases when we must use require.
-        'prefer-import/prefer-import-over-require': 1,
+        'import/no-commonjs': 1,
+        'import/no-amd': 1,
+        'import/no-dynamic-require': [1, { 'allowConditionalRequire': false }],
+        'import/first': 1,
+        // Default order is: ["builtin", "external", "parent", "sibling", "index"]
+        // https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/order.md
+        'import/order': [1, { 'alphabetize': { 'order': 'asc' } }],
```

Comment:
I'm not sure if we need to order alphabetically. What do others think?

Reply:
My thought while applying alphabetical order was to make import sections with always the same order in all the files, otherwise, we will see different orders within the same import groups.
Examples:


```js
// script1.js
import path from 'path';
import os from 'os';
import crypto from 'crypto';

// script2.js
import crypto from 'crypto';
import path from 'path';
import os from 'os';
```

```js
// component1.js
import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

//component2.js
import classnames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';
```

but like I said, it comes with the caveat of not complying with our own understanding of "importance of imports".

Reply:
Alphabetical order also comes with another improvement to the order.

This would not be valid because  the dots `..` will come before any letter:
```js
import foo from '../foo';
import qux from '../../foo/qux';
```

This way we can sort imports by folder depth as well:
```js
import qux from '../../foo/qux';
import foo from '../foo';
```

Reply:
I don't really mind if we leave the enforcement of alphabetical sorting out of equation for now precisely due to the remarks @zebateira made about the importance of imports.